factory/vm/sovereignVmContainerShardCreator.go

```diff
@@ -0,0 +1,84 @@
+package vm
+
+import (
+	"github.com/multiversx/mx-chain-go/process"
+	"github.com/multiversx/mx-chain-go/process/factory"
+	"github.com/multiversx/mx-chain-go/process/factory/shard"
+	"github.com/multiversx/mx-chain-go/process/smartContract/hooks"
+	"github.com/multiversx/mx-chain-go/process/smartContract/hooks/counters"
+	"github.com/multiversx/mx-chain-go/state/syncer"
+)
+
+type sovereignVmContainerShardFactory struct {
+	blockChainHookHandlerCreator hooks.BlockChainHookHandlerCreator
+}
+
+// NewSovereignVmContainerShardFactory creates a new sovereign vm container shard factory
+func NewSovereignVmContainerShardFactory(bhhc hooks.BlockChainHookHandlerCreator) (*sovereignVmContainerShardFactory, error) {
+	return &sovereignVmContainerShardFactory{
+		blockChainHookHandlerCreator: bhhc,
+	}, nil
+}
+
+// CreateVmContainerFactoryShard will create a new instance of sovereign vm container and factory for shard
+func (svcsf *sovereignVmContainerShardFactory) CreateVmContainerFactoryShard(argsHook hooks.ArgBlockChainHook, args ArgsVmContainerFactory) (process.VirtualMachinesContainer, process.VirtualMachinesContainerFactory, error) {
+	blockChainHookImpl, err := svcsf.blockChainHookHandlerCreator.CreateBlockChainHookHandler(argsHook)
```

Comment:
Actually, sovereign container should hold internally both shard+meta creators(from constructor, as pointers)
Inside this func it should call:
1. shard.Create...
2. meta.Create...